components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,346 @@
+                result.getEndpoint().getAuthentication().getType()
+        );
+    }
+
+    @Test(priority = 7)
+    public void testDeactivateAction() throws ActionMgtException {
+
+        Assert.assertEquals(Action.Status.ACTIVE, action.getStatus());
+        Action deactivatedAction = daoImpl.deactivateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), TENANT_ID);
+        Assert.assertEquals(Action.Status.INACTIVE, deactivatedAction.getStatus());
+    }
+
+    @Test(priority = 8)
+    public void testActivateAction() throws ActionMgtException {
+
+        Action result = daoImpl.activateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), TENANT_ID);
+        Assert.assertEquals(Action.Status.ACTIVE, result.getStatus());
+    }
+
+    @Test(priority = 9)
+    public void testUpdateActionEndpointAuthProperties() {
+
+        try {
+            Action result = daoImpl.updateActionEndpointAuthProperties(action.getId(), authType, TENANT_ID);
+            Assert.assertEquals(AuthType.AuthenticationType.BASIC, result.getEndpoint().getAuthentication().getType());
+            Assert.assertEquals(2, result.getEndpoint().getAuthentication().getProperties().size());
+        } catch (Exception e) {
+            Assert.fail();
+        }
+    }
+
+    @Test(priority = 10)
+    public void testUpdateActionEndpoint() throws ActionMgtException {
+
+        EndpointConfig endpointConfig = new EndpointConfig.EndpointConfigBuilder()
+                .uri("https://template.com")
+                .authentication(new AuthType.AuthTypeBuilder()
+                        .type(AuthType.AuthenticationType.BEARER)
+                        .properties(Arrays.asList(
+                                new AuthProperty.AuthPropertyBuilder()
+                                        .name(AuthType.AuthenticationType.AuthenticationProperty.ACCESS_TOKEN.getName())
+                                        .value("c7fce95f-3f5b-4cda-8bb1-4cb7b3990f83")
+                                        .isConfidential(AuthType.AuthenticationType.AuthenticationProperty
+                                                .ACCESS_TOKEN.getIsConfidential())
+                                        .build()))
+                        .build())
+                .build();
+        Action result = daoImpl.updateActionEndpoint(
+                PRE_ISSUE_ACCESS_TOKEN, action.getId(), endpointConfig, action.getEndpoint()
+                .getAuthentication(), TENANT_ID);
+        Assert.assertNotEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(AuthType.AuthenticationType.BEARER, result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 11)
+    public void testGetActionsCountPerType() throws ActionMgtException {
+
+        Map<String, Integer> actionMap = daoImpl.getActionsCountPerType(TENANT_ID);
+        for(Map.Entry<String, Integer> map: actionMap.entrySet()){
```

Comment:
```suggestion
        for(Map.Entry<String, Integer> entry: actionMap.entrySet()){
```
Let's use a meaningful word. map gives a wrong idea

Reply:
Fixed in  commit 441362715e216059fb4f1650ed0f1e2691ead3ab